game/dancegraph.cc

```diff
@@ -171,16 +172,16 @@ void DanceGraph::finalizeTrackChange() {
 	m_gamingMode = m_curTrackIt->first;
 	std::string gm = m_gamingMode;
 	if (gm == "dance-single") { m_pads = 4; std::copy(mapping4, mapping4+max_panels, m_arrow_map); }
-	else if (gm == "dance-double") { m_pads = 8; std::copy(mapping8, mapping8+max_panels, m_arrow_map); }
-	else if (gm == "dance-couple") { m_pads = 8; std::copy(mapping8, mapping8+max_panels, m_arrow_map); }
-	else if (gm == "dance-solo") { m_pads = 6; std::copy(mapping6, mapping6+max_panels, m_arrow_map); }
-	else if (gm == "pump-single") { m_pads = 5 ; std::copy(mapping5, mapping5+max_panels, m_arrow_map); }
-	else if (gm == "pump-double") { m_pads = 10; std::copy(mapping10, mapping10+max_panels, m_arrow_map); }
-	else if (gm == "pump-couple") { m_pads = 10; std::copy(mapping10, mapping10+max_panels, m_arrow_map); }
-	else if (gm == "ez2-single") { m_pads = 5; std::copy(mapping5, mapping5+max_panels, m_arrow_map); }
-	else if (gm == "ez2-double") { m_pads = 10; std::copy(mapping10, mapping10+max_panels, m_arrow_map); }
-	else if (gm == "ez2-real") { m_pads = 7; std::copy(mapping7, mapping7+max_panels, m_arrow_map); }
-	else if (gm == "para-single") { m_pads = 5; std::copy(mapping5, mapping5+max_panels, m_arrow_map); }
+	else if (gm == "dance-double") { m_pads = 4; std::copy(mapping8, mapping8+max_panels, m_arrow_map); }
+	else if (gm == "dance-couple") { m_pads = 4; std::copy(mapping8, mapping8+max_panels, m_arrow_map); }
+	else if (gm == "dance-solo") { m_pads = 4; std::copy(mapping6, mapping6+max_panels, m_arrow_map); }
+	else if (gm == "pump-single") { m_pads = 4; std::copy(mapping5, mapping5+max_panels, m_arrow_map); }
+	else if (gm == "pump-double") { m_pads = 4; std::copy(mapping10, mapping10+max_panels, m_arrow_map); }
+	else if (gm == "pump-couple") { m_pads = 4; std::copy(mapping10, mapping10+max_panels, m_arrow_map); }
+	else if (gm == "ez2-single") { m_pads = 4; std::copy(mapping5, mapping5+max_panels, m_arrow_map); }
+	else if (gm == "ez2-double") { m_pads = 4; std::copy(mapping10, mapping10+max_panels, m_arrow_map); }
+	else if (gm == "ez2-real") { m_pads = 4; std::copy(mapping7, mapping7+max_panels, m_arrow_map); }
+	else if (gm == "para-single") { m_pads = 4; std::copy(mapping5, mapping5+max_panels, m_arrow_map); }
```

Comment:
Something funky going on with assigning m_pads value to always 4?

Reply:
Rofl this might explain the bug @ooshlablu saw. It's evidently a botched regex (I changed it back and forth between float and unsigned like 10 times)